### PR TITLE
refactor: make transaction flag mapping more maintainable

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR makes transaction flag mapping more maintainable and fixes the tests so they actually run properly. The `XChainModifyBridge` validations were also fixed, due to catching a bug while writing tests for this PR.

This process also caught a bug with the `NFTokenCreateOffer` flag names.

### Context of Change

When reviewing #663 I looked at how transaction flags were processed, which made me realize that the `XChainModifyBridge` flags were missed. Then I wanted to make it more maintainable, to avoid this problem in the future (I didn't know this was a place I needed to update).

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

### Did you update CHANGELOG.md?

- [x] Yes

## Test Plan

Existing tests pass. Added a test for `XChainModifyBridge` flags.

### CODE REVIEW

File: tests/unit/models/transactions/test_better_transaction_flags.py [hunk 1]
```diff
@@ -49,20 +50,18 @@ def test_account_set_flags(self):
     def test_nftoken_create_offer_flags(self):
         actual = models.NFTokenCreateOffer(
             account=ACCOUNT,
-            nftoken_id="\
-                000100001E962F495F07A990F4ED55ACCFEEF365DBAA76B6A048C0A200000007",
+            nftoken_id=NFTOKEN_ID,
             amount="1000000",
             flags=models.NFTokenCreateOfferFlagInterface(
-                tf_sell_token=True,
+                TF_SELL_NFTOKEN=True,
```
Comment (mvadari) [inline tests/unit/models/transactions/test_better_transaction_flags.py:56]: Noting this as originally having the wrong name

Comment (mvadari) [inline tests/unit/models/transactions/test_better_transaction_flags.py:56]: Noting this as originally having the wrong name

**Review by khancode — APPROVED** [review]
LGTM

File: xrpl/models/flags.py [hunk 2]
```diff
@@ -1,110 +1,74 @@
...
+    # Key is transaction type name, value is mapping of flag name to int value
+    all_tx_flags: Dict[str, Dict[str, int]] = {
+        # The `:-4` here removes the `Flag` at the end of the class type to just get
+        # the transaction type name
+        **{f[:-4]: _flag_enum_to_dict(getattr(transactions, f)) for f in flag_enums},
+        **{
+            f[:-4]: _flag_enum_to_dict(getattr(transactions.pseudo_transactions, f))
+            for f in pseudo_tx_flag_enums
+        },
+    }
```
Comment (khancode) [inline xrpl/models/flags.py:57]:
```suggestion
    tx_flag_enums = [
        f for f in transactions.__all__ if f.endswith("Flag") and "Asf" not in f
    ]
    pseudo_tx_flag_enums = [
        f for f in transactions.pseudo_transactions.__all__ if f.endswith("Flag")
    ]

    # Key is transaction type name, value is mapping of flag name to int value
    all_tx_flags: Dict[str, Dict[str, int]] = {
        # The `:-4` here removes the `Flag` at the end of the class type to just get
        # the transaction type name
        **{f[:-4]: _flag_enum_to_dict(getattr(transactions, f)) for f in tx_flag_enums},
        **{
            f[:-4]: _flag_enum_to_dict(getattr(transactions.pseudo_transactions, f))
            for f in pseudo_tx_flag_enums
        },
    }
```
Renaming `flag_enums` to `tx_flag_enums` makes it specific.

**Review by JST5000 — APPROVED** [review]
LGTM

**Pull request merged**